enhancements/1845-nfd-image-compatibility/README.md

```diff
@@ -0,0 +1,184 @@
+
+    - **\<NFD label\>** - *string key value pair*.  
+    This REQUIRED property describes an image requirement against the running host OS and hardware.  
+    The value accepts: range and a literal value.
+
+  - **tags** - *array of string*  
+  This OPTIONAL property allows to group compatibility sets by tags.
+
+  - **description** - *string*  
+  This OPTIONAL property is used for a short description on a compatibility set.
+
+Example
+
+```json
+{
+  "compatibilities": [
+    {
+      "labels": {
+        "cpu-model.vendor_id": "GenuineIntel",
+        "cpu-cpuid.AVX512": true,
+        "kernel-selinux.enabled": true,
+        "kernel-config.PREEMPT": true,
+        "kernel-version.full": ">=4.19,<5.16; >=5.23",
+        "custom.glibc": ">=2.31,<=2.37"
+      },
+      "tags": ["intel"]
+    },
+    {
+      "labels": {
+        "cpu-model.vendor_id": "AuthenticAMD",
+        "cpu-cpuid.FPHP": true,
+        "kernel-selinux.enabled": true,
+        "kernel-config.PREEMPT": true,
+        "kernel-version.full": ">=4.19,<5.16; >=5.23",
+        "pci-1002_67ff.present": true,
+        "custom.glibc": ">=2.31,<=2.37"
+      },
+      "tags": ["amd"],
+      "description": "works only with AMD CPU and AMD GPU"
+    }
+  ]
+}
+```
+
+##### Discovery
+
+The subject field shall be used to associate the compatibility artifact with the target image.
+The referrers API should be used to discover artifacts.
+If one image has multiple artifacts, it is up to a client to choose the correct one.
+As a default behavior, it is recommended to get the most recent over "created" timestamp.
+
+#### NFD Key Value Notation
+
+NFD key on the image side should accept ranges and literal values. The following notation should be implemented (to describe ranges a standard comparison operators must be used):
+
+- `>, <, <=, >=`
+- `string`
+- ranges must be separated by semicolon character `;`
+
+Examples:
+
+- `"kernel-version.full": ">=4.15,<5.30"` - greater or equal to 4.15 and lower than 5.30.
+- `"kernel-version.full": ">=4.15,<5.25; >5.30"` - greater or equal to 4.15 and lower than 5.25 or greater than 5.30.
+- `"kernel-version.full": ">=5.15.x-x-generic"` - great or equal to 5.15 generic kernel flavor.
+- `"kernel-version.full": ">=5.15.x-x-generic; >=5.30"` - great or equal to 5.15 generic kernel flavor or greater or equal to any 5.30 kernel version.
+- `"kernel-version.full": "5.15"` - must equal to 5.15.
+
+#### NFD client
+
+A new standalone cmd line utility should be implemented for the NFD project that shares the same functionality as [nfd kubectl plugin](https://nfd.sigs.k8s.io/usage/kubectl-plugin).
+
+Both clients should implemented the following commands:
+
+- `validate` -  validate a NodeFeatureRule object (implemented in kubectl plugin).
+- `test` - test a NodeFeatureRule object against a node (implemented in kubectl plugin).
+- `dryrun` - DryRun a NodeFeatureRule object against a NodeFeature file (implemented in kubectl plugin).
```

Comment:
Can you better explain "dryrun" ? I'm used to seeing it as `--dryrun` to supplement a more clearly obvious action.

Reply:
I am also used to see it as you. I kept it that way to be compatibile with the [kubectl `dryrun` command](https://github.com/kubernetes-sigs/node-feature-discovery/blob/master/cmd/kubectl-nfd/subcmd/dryrun.go).